GeneratorInterface/RivetInterface/plugins/GenParticles2HepMCConverter.cc

```diff
@@ -109,27 +111,38 @@ void GenParticles2HepMCConverter::produce(edm::Event& event, const edm::EventSet
 
     hepmc_particles.push_back(hepmc_particle);
     genCandToHepMCMap[p] = hepmc_particle;
+
+    // Find incident proton pair
+    if ( p->pdgId() == 2212 and std::abs(p->eta()) > 100 and std::abs(p->pz()) > 1000 ) {
```

Comment:
hi @jhgoh - sorry for the naive question - does the hepmc listing not contain parentage information that could be used instead?

Reply:
Due to the pruning in the MiniAOD, some decay products become orphan, sometimes they appear at the front part of the event record. So "replacing" those criteria can introduce ambiguity.
But introducing additional requirements could be better to clarify the code.